ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -897,13 +921,18 @@ mkSnocket state tr = Snocket { getLocalAddr
                     Just conn@Connection { connState = SYN_SENT } -> do
                       let fd_' = FDConnected connId conn
                       writeTVar fdVarLocal fd_'
-                      writeTBQueue queue
-                                   ChannelWithInfo
-                                     { cwiAddress       = localAddress connId
-                                     , cwiSDUSize       = biSDUSize bearerInfo
-                                     , cwiChannelLocal  = connChannelRemote conn
-                                     , cwiChannelRemote = connChannelLocal conn
-                                     }
+                      when (localAddress connId /= remoteAddress) $
+                        -- We are connecting to onself.  On Linux, such
+                        -- connections are never accepted, instead the `connect`
+                        -- returns a socket which is self connected, see
+                        -- `mkConnection`.
```

Comment:
The code doesn't match the comment.
To me it looks like the case where we are _not_ connected to our self.

Reply:
Right, the comment should be removed.